mlir/tools/mlir-miopen-lib/mlir-miopen-lib.cpp

```diff
@@ -47,6 +47,10 @@ struct MiirHandle_s {
     static MLIRContext context(getRegistry());
     static std::once_flag once;
     std::call_once(once, []() {
+      // FIXME: A conv op would ignore this context and instead use the
+      // context from the UnknownLoc
+      context.printOpOnDiagnostic(false);
+      context.printStackTraceOnDiagnostic(false);
```

Comment:
@jerryyin could you explain what the current symptom is with potentially different `MLIRContext` instance in use?

Reply:
Other notes: there's one context per thread, and MLIR can spin up multiple threads to handle rewrites in parallel.

Reply:
`emitError` or `emitOpError` (that we use, or populate by a pattern failure) both shall end up in `Operation::emitError()` (refer to `mlir/lib/IR/Operation.cpp`) and should not populate an error message when `printOpOnDiagnostic()` is turned off however it still does even after I have the setting here.

Either I have misunderstood how `emitError()` work or this is a different bug somewhere. But either way the side effect is just print out a couple of lines of error message which is much less critical than not returned the error code.

Reply:
printOpOnDiagnostic() is, IIRC, something that controls whether the operation itself is printed next to the error.

Reply:
@krzysz00 So maybe it does end up to work properly. This is in terms that it only yields an error message but didn't print the op. Thanks for the the comment

Reply:
Yeah

I think what you want to do instead is to set up a diagnostic handler that swallows the error around the lowering calls that are in the library.

Reply:
And then to get rid of this bit

Reply:
@krzysz00 Sounds like a follow up ticket I'll document: This handler should print out everything when used from our internal tool like `mlir-miopen-driver` but swallow the error when used by `mlir-miopen-lib`.

Reply:
The solution is at https://mlir.llvm.org/docs/Diagnostics/

Reply:
I'd like to see it applied now, especially if we're backporting to a release

Reply:
@krzysz00 I'm not sure the commit would be backported. But yah, it sounds fair to fix the diagnostic in the same PR. Please un-approve to wait for my upcoming fix.

Reply:
Done in https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/commit/72323fdad1b523f61d1396f8fc875badab6c9aa1